pkg/controller/appdefinition/deploy.go

```diff
@@ -522,6 +524,18 @@ func addPrometheusAnnotations(annotations map[string]string, container v1.Contai
 	}
 }
 
+func addCiliumAnnotations(annotations map[string]string, container v1.Container) {
+	var visibilityConfigs []string
+	for _, port := range container.Ports {
+		if port.Protocol == v1.ProtocolHTTP {
+			visibilityConfigs = append(visibilityConfigs, fmt.Sprintf("<Ingress/%d/TCP/HTTP>", port.TargetPort))
+		}
+	}
+	if len(visibilityConfigs) > 0 {
+		annotations[labels.CiliumProxyVisibility] = strings.Join(visibilityConfigs, ",")
```

Comment:
What are the chance that this exceeds 253 characters?

Reply:
@thedadams Isn't 253 the character limit of the annotation prefix only?
I'm not aware on a limit for the annotation value :thinking: 
I just know that the total size of all keys and values must not exceed  256 KB.

Reply:
I just tested this with like 15 ports defined on one container (each is five digits long) and the annotation is created properly with no issues.